### PR TITLE
Init Manim Notebook walkthrough

### LINKED ISSUE #40: Provide an extension walkthrough for new users
[Walkthroughs](https://code.visualstudio.com/api/references/contribution-points#contributes.walkthroughs) could greatly help new users get accustomed with what our extension excepts from them and to also introduce them to the commands available. For example:

- Detect if users have already installed `manimgl`, if not show them a link to the Manim documentation.
- Add a simple sample file with two `Manim Cells` to let the explore the syntax `##`.
- Show how to preview a cell (also tell them that there are shortcuts ;))
- Show them how to use the most important commands.
- At the end, link to our documentation if they want to go more in-depth.

The walkthrough should be self-contained and only feature the highlights, otherwise we just overwhelm users. I estimate that this is going to be one of a bigger PR. Probably more something for the long-run.

### PR DESCRIPTION
**Pull request by Splines**
Closes #40

### Description

We add a new **walkthrough** to welcome new users to our extension and to accompany them in their first steps.

And as noted in [my comment here](https://github.com/Manim-Notebook/manim-notebook/issues/40#issue-2616587655):

> The walkthrough should be self-contained and only feature the highlights, otherwise we just overwhelm users.

image

I've also created some SVGs in Figma that even feature some clickable areas... Try to find them ;)

### Known limitations: Color theme

The [Walkthrough best practices](https://code.visualstudio.com/api/ux-guidelines/walkthroughs) link to a Figma plugin called [`Visual Studio Color Code Mapper`](https://www.figma.com/community/plugin/1218260433851630449). This allows to use vscode internal CSS styles in the SVG such that it adapts to the user theme. I've experimented a lot with it, but the wizard that automatically assigns the correct CSS names doesn't work greatly. I.e. previewing in other color themes reveals that many elements were _not_ mapped correctly and then suddenly a `def` keyword doesn't have enough contrast in front of the editor background.

So one would have to manually find out all the colors, e.g. color for python functions, parentheses etc. and that is just soo much work to do which I'd rather invest in developing new features or fixing bugs.

### CODE REVIEW

**Comment by bhoov** [comment]
Nice stuff! Some feedback:

1. `Check Manim Version` button, when clicked, prints the ManimGL version to the terminal and "checks" the checkbox. However, if your terminal is not open, clicking the button does not show you the version. Is there a way to print the version from the terminal in the walkthrough itself?
2. "Start with an example" opens an un-named, un-saved python file. The "preview" command doesn't work right out of the box. I see you have marked the exclamations at the top -- is there no easy way to automate the saving to some `tmp` directory? No stress if not
3. Let's add some useful shortcuts to the comments in cells. See the example commit for `Cmd+' Cmd+e` (this may be all we need, the rest can be referenced from a README and as the next step of the walkthrough)

FYI, I'm getting a bug in the 3rd cell "Make it red and fly away". `AttributeError: 'dict' object has no attribute 'frame'`, which could be fixed by instead calling `self.frame.animate.set_width(25)`. I'm guessing this is a breaking change in my manimgl==v1.7.1

Otherwise, this looks awesome! Let's get stuff in that wiki :)

**Comment by Splines** [comment]
Thanks for your review!

> 1. However, if your terminal is not open, clicking the button does not show you the version.

For me, even if the terminal is hidden, when clicking on the `Check Manim version` button it will be opened. Is that not the case for you? In the respective function, I call `const terminal = window.createTerminal("Manim Version");` and that opens the terminal for me.

> Is there a way to print the version from the terminal in the walkthrough itself?

Not in the walkthrough itself, but we could as an example show a notification with the version. However, I didn't yet implement the version parsing in the background. But apparently, it would be useful to know the Manim version also in other places, so I will implement that in another PR and for now leave the behavior here as is (with terminal opening). Opened #97 for that.

> 2. "Start with an example" opens an un-named, un-saved python file. 

Yes, because I thought it would be a bit weird if Manim would be executed for a file at some temporary location. But this is indeed a good point if users might directly click on the CodeLens without reading the comment beforehand first. I've added a new issue for this #96 and will fix this in another PR.

> 3. Let's add some useful shortcuts to the comments in cells.

Thanks, good idea!

---

By the way, have you also tested the commands that you can trigger by clicking on respective areas in the SVGs featured in the showcase?

File: src/walkthrough/sample_scene.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        # You can also preview the cell with the default hotkey:
+        # `Cmd+' Cmd+e` (MacOS) or `Ctrl+' Ctrl+e` (Windows/Linux)
+        square = Square()
+        self.play(ReplacementTransform(circle, square))
+        self.wait(1.5)
+
+        ## Make it red and fly away
+        self.play(
+            square.animate.set_fill(RED_D, opacity=0.5),
+            self.camera_config.frame.animate.set_width(25),
```
Comment (Splines) [inline src/walkthrough/sample_scene.py:28]: bhoov wrote:

> FYI, I'm getting a bug in the 3rd cell "Make it red and fly away". AttributeError: 'dict' object has no attribute 'frame', which could be fixed by instead calling self.frame.animate.set_width(25). I'm guessing this is a breaking change in my manimgl==v1.7.1

Comment (Splines) [reply src/walkthrough/sample_scene.py:28]: Mmh, good point, will check that... At the moment, with the latest Manim commit 3h ago (bafea89ac9d11dc168bce923a65501ec74312afe) and the latest VSCode version `1.96.0`, previewing doesn't work for me at all, i.e. the cells are not copied to the clipboard anymore. Will investigate in the next days...

Comment (Splines) [reply src/walkthrough/sample_scene.py:28]: I figured that my clipboard timeout setting of `850ms` was not sufficient anymore.  And it feels like the `checkpoint_paste()` command takes longer now to start which could be linked to recent refactorings that 3b1b made. Do you experience the same behavior?

Comment (Splines) [reply src/walkthrough/sample_scene.py:28]: Or maybe that's just a figment. I've just checked out several commits of Manim of the past and for all of them it takes a similar amount of time to print anything to the console after a `checkpoint_paste()`. Maybe it's just that I've worked too long on the `reload()` command which is almost instantaneous for me ;)

**Comment by bhoov** [comment]
> For me, even if the terminal is hidden, when clicking on the Check Manim version button it will be opened. Is that not the case for you?

Correct, when I click that button the terminal does not open. See the log below -- neither click opened the terminal to show the manim version.

```
2024-12-12 09:24:50.581 [info] [logger.ts:72:16] [unknown] 📜 Logfile found and cleared at 2024-12-12T17:24:50.580Z
2024-12-12 09:24:52.425 [info] [logger.ts:236:16] [unknown] 📜 Logfile recording started
2024-12-12 09:24:52.425 [info] [logger.ts:76:16] [unknown] Operating system: Darwin 23.4.0 arm64
2024-12-12 09:24:52.425 [info] [logger.ts:77:16] [unknown] Process versions: {"node":"20.18.0","acorn":"8.12.1","ada":"2.9.0","ares":"1.33.1","base64":"0.5.2","brotli":"1.0.9","cjs_module_lexer":"1.4.1","cldr":"44.1","icu":"74.2","llhttp":"8.1.2","modules":"128","napi":"9","nghttp2":"1.61.0","openssl":"0.0.0","simdutf":"5.5.0","tz":"2024a","undici":"6.19.8","unicode":"15.1","uv":"1.46.0","uvwasi":"0.0.21","v8":"12.8.374.38-electron.0","zlib":"1.3.0.1-motley","electron":"32.2.1","chrome":"128.0.6613.186","microsoft-build":"10427718"}
2024-12-12 09:24:52.425 [info] [logger.ts:82:20] [unknown] Manim notebook version: 0.0.4
2024-12-12 09:24:52.425 [info] [logger.ts:93:16] [unknown] --------------------------
2024-12-12 09:24:53.686 [info] [commands.ts:11:20] [unknown] 💠 Command Check Manim Version requested
2024-12-12 09:24:54.342 [info] [commands.ts:11:20] [unknown] 💠 Command Check Manim Version requested
2024-12-12 09:24:55.247 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
mmanimgl --version

133;C;633;E;manimgl --version;363e7287-4376-4abd-8e09-4746b358b400633;C
2024-12-12 09:24:55.906 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
mmanimgl --version

133;C;633;E;manimgl --version;1efd524c-6b3a-4e83-8f12-cf49f7c32bc1633;C
2024-12-12 09:24:56.311 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
ManimGL v1.7.1

2024-12-12 09:24:56.311 [debug] [manimShell.ts:683:36] [unknown] 👋 Manim detected in new terminal, exiting old scene
2024-12-12 09:24:56.311 [debug] [manimShell.ts:453:20] [unknown] 🔚 Force-quitting active shell
2024-12-12 09:24:56.311 [debug] [manimShell.ts:656:16] [unknown] 💨 Sending keyboard interrupt to terminal
2024-12-12 09:24:56.324 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:


%                                                                               
 
133;C;133;D;1301337;RemoteHost=hoo@Hoo-M11337;CurrentDir=/Users/hoo/Projects/manim633;E;;de5140b4-11f3-4fad-b12e-1c5e33b6fbf1633;C
2024-12-12 09:24:56.325 [debug] [manimShell.ts:756:28] [unknown] 🔚 Active shell execution ended, will reset
2024-12-12 09:24:56.325 [debug] [manimShell.ts:428:16] [unknown] 💫 Reset active shell
2024-12-12 09:24:56.563 [debug] [manimShell.ts:428:16] [unknown] 💫 Reset active shell
2024-12-12 09:24:56.564 [debug] [manimShell.ts:686:32] [unknown] 👋 Manim welcome string detected
2024-12-12 09:24:56.564 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
%                                                                               
 
133;D;01337;RemoteHost=hoo@Hoo-M11337;CurrentDir=/Users/hoo/Projects/manim633;D;0633;P;Cwd=/Users/hoo/Projects/manim
2024-12-12 09:24:56.903 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
ManimGL v1.7.1

2024-12-12 09:24:56.903 [debug] [manimShell.ts:683:36] [unknown] 👋 Manim detected in new terminal, exiting old scene
2024-12-12 09:24:56.903 [debug] [manimShell.ts:453:20] [unknown] 🔚 Force-quitting active shell
2024-12-12 09:24:56.903 [debug] [manimShell.ts:656:16] [unknown] 💨 Sending keyboard interrupt to terminal
2024-12-12 09:24:56.915 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:


%                                                                               
 
133;C;133;D;1301337;RemoteHost=hoo@Hoo-M11337;CurrentDir=/Users/hoo/Projects/manim633;E;;363e7287-4376-4abd-8e09-4746b358b400633;C
2024-12-12 09:24:56.917 [debug] [manimShell.ts:756:28] [unknown] 🔚 Active shell execution ended, will reset
2024-12-12 09:24:56.917 [debug] [manimShell.ts:428:16] [unknown] 💫 Reset active shell
2024-12-12 09:24:57.158 [debug] [manimShell.ts:428:16] [unknown] 💫 Reset active shell
2024-12-12 09:24:57.158 [debug] [manimShell.ts:686:32] [unknown] 👋 Manim welcome string detected
2024-12-12 09:24:57.158 [trace] [manimShell.ts:678:28] [unknown] 🧾 Terminal data:
%                                                                               
 
133;D;01337;RemoteHost=hoo@Hoo-M11337;CurrentDir=/Users/hoo/Projects/manim633;D;0
2024-12-12 09:25:00.084 [info] [extension.ts:65:11] [unknown] 💠 Command requested: Finish Recording Log File
```

> Opened https://github.com/Manim-Notebook/manim-notebook/issues/97 for that. 
👍


> have you also tested the commands that you can trigger by clicking on respective areas in the SVGs featured in the showcase?

Yes it works as expected! Tabs may only be meaningful if there are multiple though 😂

File: src/walkthrough/sample_scene.py (hunk 1 above)
Comment (bhoov) [reply src/walkthrough/sample_scene.py]: Oh weird, the example is working totally fine for me. 

ManimGL v1.7.1
VSCode v1.95.0

**Review by bhoov — APPROVED** [review]
Everything works on my end with my VSCode and manimgl versions. I will approve this PR. @Splines when things are working for you, feel free to merge

**Comment by Splines** [comment]
> Everything works on my end with my VSCode and manimgl versions. I will approve this PR

yeah

>  Yes it works as expected! Tabs may only be meaningful if there are multiple though 😂

😅

> Interesting thing to note -- I also use Cursor and it looks like Cursor does not support the walkthrough.

Sorry, but what is Cursor? Is it [this editor](https://www.cursor.com/)? If yes, what's its relation to VSCode?

**Pull request merged**